src/test/streams_tests.cpp

```diff
@@ -29,7 +29,18 @@ BOOST_AUTO_TEST_CASE(xor_file)
         BOOST_CHECK_EXCEPTION(xor_file.ignore(1), std::ios_base::failure, HasReason{"AutoFile::ignore: file handle is nullpt"});
     }
     {
-        AutoFile xor_file{raw_file("wbx"), xor_pat};
+#ifdef __MINGW64__
+        // The MinGW-w64 toolchain links executables to the old msvcrt C Runtime
+        // Library that does not support the `x` modifier for the _wfopen()
+        // function. It is possible to link to the new UCRT library instead.
+        // However, the most easiest way to do it is to use a new GCC with
+        // the -mcrtdll=ucrt option, which has been available since the commit
+        // https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=453cb585f0f8673a5d69d1b420ffd4b3f53aca00
```

Comment:
I think this comment could just be something like `Our usage of mingw-w64 and the msvcrt runtime does not support the x modifier for the _wfopen()`. The other info seems a bit overkiil, I'm also not sure if it's correct. From a quick look, if you choose the runtime you want, when you compile mingw-w64, there's no need for any GCC flag.

Reply:
> If the new runtime is available in guix, I'd say it would be better to just use it, instead of patching the code. Or is there some downside I am missing?

I agree that we should move to the new runtime. I assume the downside is possibly dropping support for Windows 7, but I don't see why that would be a problem.

Reply:
> if you choose the runtime you want, when you compile mingw-w64

Yes, there is `--with-default-msvcrt=ucrt` configuration option.

Reply:
Yea. So I don't see why anything GCC related is required, or needs to be mentioned here.